Frontend build fails: missing lib/api.ts and lib/types.ts

## Problem

The frontend fails to build with the error:
```
Module not found: Can't resolve '@/lib/api'
```

This causes the \"Error calculating aggregate impact / network error\" message that users see when trying to use the calculator.

## Root Cause

The `frontend/lib/api.ts` and `frontend/lib/types.ts` files are referenced in the codebase but never existed:
- `frontend/hooks/useAggregateImpact.ts` imports from `@/lib/api`
- `frontend/hooks/useHouseholdImpact.ts` imports from `@/lib/api`
- Multiple components import types from `@/lib/types`

The README even documents these files as existing, but they were never created.

## Impact

- Frontend build fails completely
- Users see \"network error\" when trying to calculate aggregate impact
- The calculator is non-functional

## Solution

Create the missing files:
- `frontend/lib/api.ts` - API client with proper timeout handling (2 min for aggregate calculations)
- `frontend/lib/types.ts` - TypeScript types matching backend Pydantic models

--- a/.gitignore
+++ b/.gitignore
@@ -11,6 +11,7 @@ downloads/
 eggs/
 .eggs/
 lib/
+!frontend/lib/
 lib64/
 parts/
 sdist/

--- a/frontend/.eslintrc.json
+++ b/frontend/.eslintrc.json
@@ -1,0 +1,1 @@
+{"extends": "next/core-web-vitals"}

--- a/frontend/app/page.tsx
+++ b/frontend/app/page.tsx
@@ -117,7 +117,7 @@ export default function Home() {
                 <h2 className="text-2xl font-bold text-primary mb-4">Get Started</h2>
                 <p className="text-gray-700 mb-4">
                   Configure your household in the sidebar, then click{' '}
-                  <span className="font-semibold">"Calculate Impact"</span> to see:
+                  <span className="font-semibold">&ldquo;Calculate Impact&rdquo;</span> to see:
                 </p>
                 <ul className="list-disc list-inside space-y-2 text-gray-700">
                   <li>How the Rhode Island Child Tax Credit would benefit your household</li>

--- a/frontend/lib/api.ts
+++ b/frontend/lib/api.ts
@@ -1,0 +1,159 @@
+/**
+ * API client for RI CTC Calculator backend.
+ */
+
+import type {
+  ReformParams,
+  HouseholdRequest,
+  HouseholdImpactResponse,
+  AggregateImpactResponse,
+  HealthResponse,
+  DatasetSummary,
+} from "./types";
+
+// API base URL from environment variable, defaults to localhost:8080
+const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:8080";
+
+// Timeout for API requests (2 minutes for aggregate calculations)
+const DEFAULT_TIMEOUT = 120000;
+
+/**
+ * Custom error class for API errors.
+ */
+export class ApiError extends Error {
+  constructor(
+    message: string,
+    public status?: number,
+    public response?: unknown,
+  ) {
+    super(message);
+    this.name = "ApiError";
+  }
+}
+
+/**
+ * Make a fetch request with timeout support.
+ */
+async function fetchWithTimeout(
+  url: string,
+  options: RequestInit,
+  timeout: number = DEFAULT_TIMEOUT,
+): Promise<Response> {
+  const controller = new AbortController();
+  const timeoutId = setTimeout(() => controller.abort(), timeout);
+
+  try {
+    const response = await fetch(url, {
+      ...options,
+      signal: controller.signal,
+    });
+    return response;
+  } finally {
+    clearTimeout(timeoutId);
+  }
+}
+
+/**
+ * API client methods.
+ */
+export const api = {
+  /**
+   * Calculate household impact across income range.
+   */
+  async calculateHouseholdImpact(
+    request: HouseholdRequest,
+  ): Promise<HouseholdImpactResponse> {
+    const response = await fetchWithTimeout(
+      `${API_BASE_URL}/api/household-impact`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(request),
+      },
+    );
+
+    if (!response.ok) {
+      const errorData = await response.json().catch(() => null);
+      throw new ApiError(
+        errorData?.detail || `HTTP error ${response.status}`,
+        response.status,
+        errorData,
+      );
+    }
+
+    return response.json();
+  },
+
+  /**
+   * Calculate statewide aggregate impact.
+   * Note: This takes ~90 seconds due to microsimulation calculations.
+   */
+  async calculateAggregateImpact(
+    reformParams: ReformParams,
+  ): Promise<AggregateImpactResponse> {
+    const response = await fetchWithTimeout(
+      `${API_BASE_URL}/api/aggregate-impact`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ reform_params: reformParams }),
+      },
+      DEFAULT_TIMEOUT,
+    );
+
+    if (!response.ok) {
+      const errorData = await response.json().catch(() => null);
+      throw new ApiError(
+        errorData?.detail || `HTTP error ${response.status}`,
+        response.status,
+        errorData,
+      );
+    }
+
+    return response.json();
+  },
+
+  /**
+   * Health check endpoint.
+   */
+  async health(): Promise<HealthResponse> {
+    const response = await fetchWithTimeout(`${API_BASE_URL}/api/health`, {
+      method: "GET",
+    });
+
+    if (!response.ok) {
+      throw new ApiError(`Health check failed: ${response.status}`);
+    }
+
+    return response.json();
+  },
+
+  /**
+   * Get dataset summary statistics.
+   */
+  async getDatasetSummary(): Promise<DatasetSummary> {
+    const response = await fetchWithTimeout(
+      `${API_BASE_URL}/api/dataset-summary`,
+      {
+        method: "GET",
+      },
+    );
+
+    if (!response.ok) {
+      const errorData = await response.json().catch(() => null);
+      throw new ApiError(
+        errorData?.detail || `HTTP error ${response.status}`,
+        response.status,
+        errorData,
+      );
+    }
+
+    return response.json();
+  },
+};
+
+export default api;

--- a/frontend/lib/types.ts
+++ b/frontend/lib/types.ts
@@ -1,0 +1,152 @@
+/**
+ * TypeScript type definitions for RI CTC Calculator.
+ * These types mirror the Pydantic models in the backend.
+ */
+
+/**
+ * Filing status phaseout thresholds.
+ */
+export interface PhaseoutThresholds {
+  SINGLE: number;
+  JOINT: number;
+  HEAD_OF_HOUSEHOLD: number;
+  SURVIVING_SPOUSE: number;
+  SEPARATE: number;
+}
+
+/**
+ * Reform parameters for RI CTC calculations.
+ */
+export interface ReformParams {
+  // CTC parameters
+  ctc_amount: number;
+  ctc_age_limit: number;
+  ctc_refundability_cap: number;
+  ctc_phaseout_rate: number;
+  ctc_phaseout_thresholds: PhaseoutThresholds;
+  ctc_young_child_boost_amount: number;
+  ctc_young_child_boost_age_limit: number;
+
+  // Dependent exemption parameters
+  enable_exemption_reform: boolean;
+  exemption_amount: number;
+  exemption_age_limit_enabled: boolean;
+  exemption_age_threshold: number;
+  exemption_phaseout_rate: number;
+  exemption_phaseout_thresholds: PhaseoutThresholds;
+}
+
+/**
+ * Request for household impact calculation.
+ */
+export interface HouseholdRequest {
+  age_head: number;
+  age_spouse: number | null;
+  dependent_ages: number[];
+  income: number;
+  reform_params: ReformParams;
+}
+
+/**
+ * Request for aggregate/statewide impact calculation.
+ */
+export interface AggregateImpactRequest {
+  reform_params: ReformParams;
+}
+
+/**
+ * Benefit breakdown at a specific income level.
+ */
+export interface BenefitAtIncome {
+  baseline: number;
+  reform: number;
+  difference: number;
+  ctc_component: number;
+  exemption_tax_benefit: number;
+}
+
+/**
+ * Response for household impact calculation.
+ */
+export interface HouseholdImpactResponse {
+  // Income sweep arrays for charts
+  income_range: number[];
+  ctc_baseline_range: number[];
+  ctc_reform_range: number[];
+
+  // Component breakdowns
+  ctc_component: number[];
+  exemption_tax_benefit: number[];
+
+  // Specific benefit at user's income
+  benefit_at_income: BenefitAtIncome;
+
+  // Chart configuration
+  x_axis_max: number;
+
+  // Diagnostic data (optional)
+  diagnostics?: Record<string, unknown>;
+}
+
+/**
+ * Impact by income bracket.
+ */
+export interface IncomeBracket {
+  bracket: string;
+  beneficiaries: number;
+  total_cost: number;
+  avg_benefit: number;
+}
+
+/**
+ * Response for aggregate/statewide impact calculation.
+ */
+export interface AggregateImpactResponse {
+  // Top-level metrics
+  total_cost: number;
+  beneficiaries: number;
+  avg_benefit: number;
+  children_affected: number;
+
+  // Winners/losers
+  winners: number;
+  losers: number;
+  winners_rate: number;
+  losers_rate: number;
+
+  // Poverty impacts
+  poverty_baseline_rate: number;
+  poverty_reform_rate: number;
+  poverty_rate_change: number;
+  poverty_percent_change: number;
+
+  child_poverty_baseline_rate: number;
+  child_poverty_reform_rate: number;
+  child_poverty_rate_change: number;
+  child_poverty_percent_change: number;
+
+  // Income bracket breakdown
+  by_income_bracket: IncomeBracket[];
+}
+
+/**
+ * Health check response.
+ */
+export interface HealthResponse {
+  status: string;
+  dataset_loaded: boolean;
+  version: string;
+}
+
+/**
+ * Dataset summary response.
+ */
+export interface DatasetSummary {
+  household_count: number;
+  person_count: number;
+  median_agi: number;
+  p75_agi: number;
+  p90_agi: number;
+  total_children: number;
+  households_with_children: number;
+}

